xdg-desktop-portal-cosmic: 1.0.0-alpha.5.1 -> 1.0.0-alpha.6

HeitorAugustoLN opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

- Remove vendored `Cargo.lock` that was forgotten to be removed
- Fix service execution
- `xdg-desktop-portal-cosmic` is now using `libcosmicAppHook`
- Modernized the package
- Added myself as one of the maintainers of the package
- Updated it from `1.0.0-alpha.5.1` to `1.0.0-alpha.6`
- Added an updateScript

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

HeitorAugustoLN committed:
xdg-desktop-portal-cosmic: use libcosmicAppHook

HeitorAugustoLN committed:
xdg-desktop-portal-cosmic: refactor

HeitorAugustoLN committed:
xdg-desktop-portal-cosmic: remove vendored Cargo.lock

It wasn't used anymore, but it didn't get deleted

HeitorAugustoLN committed:
xdg-desktop-portal-cosmic: add updateScript

HeitorAugustoLN committed:
xdg-desktop-portal-cosmic: 1.0.0-alpha.5.1 -> 1.0.0-alpha.6

HeitorAugustoLN committed:
xdg-desktop-portal-cosmic: add HeitorAugustoLN as a maintainer

HeitorAugustoLN committed:
xdg-desktop-portal-cosmic: fix service execution

The service file was configured to use /bin/false which has two issues:
1. /bin
2. The service intentionally fails to start

a-kenji reviewed (approved):
Not happy about the usage of the `libcosmicAppHook` without proper documentation for it first akin to how the [Gnome Hooks](https://nixos.org/manual/nixpkgs/stable/#ssec-gnome-hooks) are documented.

So I personally prefer that being done first, or at least in a follow up PR.

The rest looks good.

HeitorAugustoLN commented:
> Not happy about the usage of the libcosmicAppHook without proper documentation for it first akin to how the [Gnome Hooks](https://nixos.org/manual/nixpkgs/stable/#ssec-gnome-hooks) are documented.
>
> So I personally prefer that being done first, or at least in a follow up PR.

I can write documentation for it tomorrow in another PR

a-kenji commented:
Just because the support got added doesn't mean everything has to be done in one PR imho. But generally that would not be a bad idea, yes.

HeitorAugustoLN commented:
> Shouldn't this switch to using finalAttrs as well since support for rust got added here: #382550 ?

Didn't know about that! That is cool, I can switch to it

HeitorAugustoLN committed:
xdg-desktop-portal-cosmic: replace `rec` with `finalAttrs`

a-kenji commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 386198`

---
### `x86_64-linux`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>cosmic-session</li>
    <li>xdg-desktop-portal-cosmic</li>
    <li>xdg-desktop-portal-cosmic.debug</li>
  </ul>
</details>

HeitorAugustoLN commented:
This is odd, it wasn't failing before

a-kenji commented:
Maybe it is a me thing? Can you try building it?

HeitorAugustoLN commented:
> Maybe it is a me thing? Can you try building it?

I did built it, it is failing now. ofborg builds is also failing

HeitorAugustoLN commented:
I got it. We will need a patch, I really don't know why it wasn't happening before.

HeitorAugustoLN committed:
xdg-desktop-portal-cosmic: fix package tests after ashpd API updates

Apply upstream patch to fix compilation errors in example code that were causing
test failures. The issues were related to API changes in the ashpd dependency

HeitorAugustoLN commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 386198`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>cosmic-session</li>
    <li>xdg-desktop-portal-cosmic</li>
    <li>xdg-desktop-portal-cosmic.debug</li>
  </ul>
</details>

JohnRTitor commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 386198`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>cosmic-session</li>
    <li>xdg-desktop-portal-cosmic</li>
    <li>xdg-desktop-portal-cosmic.debug</li>
  </ul>
</details>
